router/core/errors.go

```diff
@@ -257,12 +263,18 @@ func writeMultipartError(w http.ResponseWriter, requestErrors graphqlerrors.Requ
 		return err
 	}
 
-	resp, err := wrapMultipartMessage(responseBytes)
+	isSubscription := operationType == "subscription"
+	resp, err := wrapMultipartMessage(responseBytes, isSubscription)
 	if err != nil {
 		return err
 	}
 
-	resp = append(resp, '\n')
+	if isSubscription {
```

Comment:
FYI: I am not sure if we actually need this, as from what I gathered we call this from the router (and not from the gotools subscription code), but I need to look into subscriptions a bit more, and see how errors are emitted for subscriptions via gotools.

Subscriptions are also broken for the apollo kotlin client,  so I will take a look at both of these together.

Reply:
This is called from `writeRequestErrors` which can be called when using subscription over SSE or multipart.

Reply:
So what I am worried about is duplicate end boundaries, so in flushwriter on complete (which is called through graphqlgotools) we write
```
_, _ = f.writer.Write([]byte("--" + multipartBoundary + "--\n"))
```
since we have this, if I remove the if condition I want to make sure that we cant have duplicate end boundaries.
That is
```
---graphql---
---graphql---
```
 Thus the if condition to be on the safe side, will verify and remove in a pr.